sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -1462,13 +1480,50 @@ export class BlobClient extends StorageClient {
    */
   public async downloadToBuffer(
     buffer: Buffer,
-    offset: number,
+    offset?: number,
     count?: number,
-    options: BlobDownloadToBufferOptions = {}
-  ): Promise<Buffer> {
+    options?: BlobDownloadToBufferOptions
+  ): Promise<Buffer>;
+
+  public async downloadToBuffer(
+    param1: Buffer | number = 0,
+    param2: number = 0,
+    param3: BlobDownloadToBufferOptions | number = 0,
+    param4: BlobDownloadToBufferOptions = {}
+  ) {
+    let buffer: Buffer;
+    let offset: number;
+    let count: number;
+    let options = param1 instanceof Buffer ? param4 : (param3 as BlobDownloadToBufferOptions);
     const { span, spanOptions } = createSpan("BlobClient-downloadToBuffer", options.spanOptions);
 
     try {
+      if (param1 instanceof Buffer) {
+        buffer = param1;
+        offset = param2;
+        count = param3 as number;
+      } else {
+        offset = param1;
+        count = param2 as number;
+        try {
+          buffer = Buffer.alloc(
+            count > 0
+              ? count
+              : (await this.getProperties({
+                  ...options,
```

Comment:
Does `BlobDownloadToBufferOptions` fits `getProperties`?

Reply:
Hmmm. 
This was being used below in this same `downloadToBuffer` method.
`options` definitely has more attributes, probably this is allowed because all the attributes are optional in `BlobGetPropertiesOptions`.

![image](https://user-images.githubusercontent.com/10452642/67429135-5dd95580-f594-11e9-8c25-00b9b52ac554.png)

Just to be safe, I'll update this to pass in the exact attributes it expects.